### ISSUE TITLE
write MSH4.1

### ISSUE THREAD

**Issue by gdmcbain** [nschloe/meshio#332]
The reading of MSH4.1 #330 is implemented in #331 but not the writing.

Writing is less pressing than reading for MSH4.1 since it is generated by Gmsh by default but not read by much except Gmsh and meshio (with #331).

**Comment by gdmcbain** [comment]
This isn't going so well.  I don't understand how to write the `$Entities` block.  The spec says that
>  The $Entities section is now optional. 

but this isn't always the case, Gmsh 4.2.0-git-0d5d49207 sometimes complaining 
> Error   : Unknown entity 2 of dimension 1

**Comment by nschloe** [comment]
Best prepare and example and file a bug over at gmsh's gitlab. Tag me, too.

**Comment by nschloe** [comment]
Fixed by #435.

**Comment by gdmcbain** [comment]
An error like that reported above:

```
Error : Unknown entity 2 of dimension 1
```

arose again, after the closing of this issue by #435, in #470:

```
Error   : Unknown entity 0 of dimension 3
```
